File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        
        if not head:
            return None
        
        dummy = ListNode(0,head)
        prev = dummy
        
        while head and head.next:
            if head.val == head.next.val:
                while head.next and head.val == head.next.val: #while duplicates are found
                    head = head.next
                head = head.next
                prev.next = head
            else:
                prev = prev.next
                head= head.next
        return dummy.next
    

        
        

File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* deleteDuplicates(struct ListNode* head){
    
    if (head ==NULL)
        return head;
        
    struct ListNode *dummy = (struct ListNode*)malloc(sizeof(struct ListNode));
    dummy->next = head;
    struct ListNode *prev = dummy;
    
    while (head != NULL && head->next != NULL)
        if (head->val == head->next->val){
            while (head->next !=NULL && head->val==head->next->val){
                                head= head->next;
            }

            head=head->next;
            prev->next = head;
            
        }else{
            
            head= head->next;
            prev=prev->next;
            
        }

    
    
    return dummy->next;
                

}